packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -161,57 +165,81 @@ export const generateTypeDefGlobalContext = () => {
 // and they may want to generate a custom side. :shrug
 // TODO: Figure out how to get a list of scalars from the api-side so that
 // they don't get out of sync.
-export const generateTypeDefGraphQL = async (
-  side: 'web' | 'api' | 'all' = 'all'
-) => {
-  const rwjsPaths = getPaths()
-  type GenerateResponse = { filename: string; contents: string }[]
+export const generateTypeDefGraphQLApi = async () => {
   try {
-    // TODO: Figure out how to get this type.
-    const generates: Record<string, unknown> = {}
-
-    if (['api', 'all'].includes(side)) {
-      generates[path.join(rwjsPaths.api.base, 'types/graphql.d.ts')] = {
-        plugins: ['typescript', 'typescript-resolvers'],
-      }
-    }
-    if (['web', 'all'].includes(side) && findCells().length) {
-      generates[path.join(rwjsPaths.web.base, 'types/graphql.d.ts')] = {
-        documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
-        plugins: ['typescript', 'typescript-operations'],
-      }
-    }
-
-    // https://www.graphql-code-generator.com/docs/getting-started/programmatic-usage#using-the-cli-instead-of-core
-    const f: GenerateResponse = await generate(
-      {
-        cwd: rwjsPaths.base,
-        schema: rwjsPaths.generated.schema,
-        config: {
-          scalars: {
-            DateTime: 'string',
-            Date: 'string',
-            JSON: 'Record<string, unknown>',
-            JSONObject: 'Record<string, unknown>',
-            Time: 'string',
+    const rwjsPaths = getPaths()
+    const f = await generateTypeDefGraphQL({
+      [path.join(rwjsPaths.api.base, 'types/graphql.d.ts')]: {
+        plugins: [
+          {
+            typescript: { enumsAsTypes: true },
           },
-          omitOperationSuffix: true, // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc.
-        },
-        // @ts-expect-error - Meh.
-        generates,
-        silent: false,
-        errorsOnly: true,
+          'typescript-resolvers',
+        ],
       },
-      true
-    )
-    return f.map(({ filename }) => filename)
+    })
+    return f
   } catch (e) {
-    // `generate` outputs errors which are helpful.
-    // This tries to clean up the output of those errors.
     console.error()
-    console.error('Error: Could not generate GraphQL type definitions')
+    console.error('Error: Could not generate GraphQL type definitions (api)')
     console.error()
+    return []
+  }
+}
 
+export const generateTypeDefGraphQLWeb = async () => {
+  if (findCells().length) {
+    const rwjsPaths = getPaths()
+    try {
+      const f = await generateTypeDefGraphQL({
+        [path.join(rwjsPaths.web.base, 'types/graphql.d.ts')]: {
+          documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
+          plugins: [
```

Comment:
I think this is missing the `omitOperationSuffix: true,`, the api one probably doens't need it, but the web one does (not 100% sure)

Reply:
It's lower down over here: https://github.com/redwoodjs/redwood/pull/2839/files#diff-e60ed2b5527fb9736c6f69a43c98c5e71372805eb6bcbe0ee01372854e81086cR235